phoenix-core/src/it/java/org/apache/phoenix/replication/ReplicationLogGroupIT.java

```diff
@@ -0,0 +1,339 @@
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            String ddl = String.format("create table %s (id1 integer not null, " +
+                    "id2 integer not null, val1 varchar, val2 varchar " +
+                    "constraint pk primary key (id1, id2))", tableName);
+            conn.createStatement().execute(ddl);
+            ddl = String.format("create index %s on %s (val1) include (val2)",
+                    indexName, tableName);
+            conn.createStatement().execute(ddl);
+            conn.commit();
+        }
+        JVMClusterUtil.RegionServerThread rs2 = cluster.startRegionServer();
+        ServerName sn2 = rs2.getRegionServer().getServerName();
+        moveRegionToServer(table, sn2);
+        moveRegionToServer(TableName.valueOf(SYSTEM_CATALOG_NAME), sn2);
+        moveRegionToServer(TableName.valueOf(SYSTEM_CHILD_LINK_NAME), sn2);
+        int rowCount = 50;
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            PreparedStatement stmt = conn.prepareStatement(
+                    "upsert into " + tableName + " VALUES(?, ?, ?, ?)");
+            // upsert 50 rows
+            for (int i = 0; i < 5; ++i) {
+                for (int j = 0; j < 10; ++j) {
+                    stmt.setInt(1, i);
+                    stmt.setInt(2, j);
+                    stmt.setString(3, "abcdefghijklmnopqrstuvwxyz");
+                    stmt.setString(4, null); // Generate a DeleteColumn cell
+                    stmt.executeUpdate();
+                }
+                // we want to simulate RS crash after updating memstore and WAL
+                IndexRegionObserver.setIgnoreSyncReplicationForTesting(true);
+                conn.commit();
+            }
+            // Create tenant views for syscat and child link replication
+            createViewHierarchy();
+        } finally {
+            IndexRegionObserver.setIgnoreSyncReplicationForTesting(false);
+        }
+        cluster.killRegionServer(rs2.getRegionServer().getServerName());
+        Threads.sleep(20000); // just to be sure that the kill has fully started.
+        util.waitUntilAllRegionsAssigned(table);
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            Map<String, Integer> expected = Maps.newHashMap();
+            // For each row 1 Put + 1 Delete (DeleteColumn)
+            expected.put(tableName, rowCount * 2);
+            // unverified + verified + delete (Delete column)
+            expected.put(indexName, rowCount * 3);
+            // 1 tenant view was created
+            expected.put(SYSTEM_CHILD_LINK_NAME, 1);
+            // atleast 1 log entry for syscat
+            expected.put(SYSTEM_CATALOG_NAME, 1);
+            verifyReplication(conn, expected);
+        }
+    }
+
+    @Test
+    public void testSystemTables() throws Exception {
+        createViewHierarchy();
+        Map<String, List<Mutation>> logsByTable = groupLogsByTable();
```

Comment:
We should have a test at for a system table that is not the catalog or child link table and show that it is not replicated.